modular/gater/object_handler.go

```diff
@@ -396,19 +397,25 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		log.CtxDebugw(reqCtx.Context(), reqCtx.String())
 	}()
 
-	queryParams := r.URL.Query()
-	gnfdUserParam := queryParams.Get(GnfdUserAddressHeader)
-	gnfdOffChainAuthAppDomainParam := queryParams.Get(GnfdOffChainAuthAppDomainHeader)
-	gnfdAuthorizationParam := queryParams.Get(GnfdAuthorizationHeader)
-
-	// if all required off-chain auth headers are passed in as query params, we fill corresponding headers
-	if gnfdUserParam != "" && gnfdOffChainAuthAppDomainParam != "" && gnfdAuthorizationParam != "" {
-		r.Header.Set(GnfdUserAddressHeader, gnfdUserParam)
-		r.Header.Set(GnfdOffChainAuthAppDomainHeader, gnfdOffChainAuthAppDomainParam)
-		r.Header.Set(GnfdAuthorizationHeader, gnfdAuthorizationParam)
-	}
-
 	reqCtx, reqCtxErr = NewRequestContext(r, g)
+	if reqCtxErr != nil {
+		queryParams := r.URL.Query()
+		gnfdUserParam := queryParams.Get(GnfdUserAddressHeader)
+		gnfdOffChainAuthAppDomainParam := queryParams.Get(GnfdOffChainAuthAppDomainHeader)
+		gnfdOffChainAuthAppExpiryTimestampParam := queryParams.Get(commonhttp.HTTPHeaderExpiryTimestamp)
+		gnfdAuthorizationParam := queryParams.Get(GnfdAuthorizationHeader)
+
+		// if all required off-chain auth headers are passed in as query params, we fill corresponding headers
+		if gnfdUserParam != "" && gnfdOffChainAuthAppDomainParam != "" && gnfdAuthorizationParam != "" && gnfdOffChainAuthAppExpiryTimestampParam != "" {
```

Comment:
It seems if `reqCtxErr != nil` and the condition here is false, we can by pass the signature check.

Reply:
Yes, but authenticated should be still kept as "false", so that private object can't be accessed in this case.
But I will try refactor according to next suggestion.